.changelog/14127.txt

```diff
@@ -0,0 +1,7 @@
+```release-note:improvement
+client: add option to restart all tasks of an allocation, even if the task already run, such as non-sidecar prestart and poststart tasks.
+```
+
+```release-note:improvement
+client: only start poststop tasks after poststart tasks are done.
+```
```

Comment:
bug or feature?

Reply:
We never specified the expected behaviour here, so it's not like it was doing something wrong. The code change will now always follow this order, so I think it' an `improvement`.